Controllers specs

Wahaj-Ali opened the pull request:
## Highlights
- Add RSpec as a gem to the project.
- Add tests for each action in the controllers-spec to check
  - If response status was correct.
  - If a correct template was rendered.
  - If the response body includes correct placeholder text.

Wahaj-Ali committed:
Install gem for controllers test and add tests for users_controller.

In this commit
- Install the gem to test the controllers using RSpec.
- Add tests fot the index of users controller.

Wahaj-Ali committed:
Create posts_controller_spec.rb

In this commit
- Add tests for the posts controller.

Wahaj-Ali committed:
Add tests for show method in users controller.

Ismailco commented on spec/controllers/users_controller_spec.rb:
```diff
@@ -0,0 +1,47 @@
+    end
+  end
+
+  describe 'GET #index' do
```
- [x] [OPTIONAL] Kindly make sure to pay attention to small details for example here the test should be `'Get #show'`, this may cause confusion in debugging if an error occurred

Wahaj-Ali replied:
Sure brother @Ismailco.

Ismailco commented on spec/controllers/posts_controller_spec.rb:
```diff
@@ -0,0 +1,37 @@
+
+    it 'renders the correct template' do
+      get user_post_url(1, 3)
+      expect(response).to render_template('show')
```
- [x] [OPTIONAL] I suggest using the same syntax in your code so that when debugging or reading the code you can easily skim through (`:show` and `'show'`)

Ismailco reviewed (approved):
### Hi @Wahaj-Ali, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![cr-approve](https://user-images.githubusercontent.com/25227022/165483662-7b06c812-4cc0-40b9-a84c-4729d4cf10c3.gif)


## Highlights

✔️ Implement Unite tests
✔️ Clean code
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Wahaj-Ali commented:
@Ismailco thank you for your valuable suggestions. I will implement the optional changes to improve my code further 😄

Wahaj-Ali committed:
Made optional changes suggested by the reviewer to improve the code

In this commit
- Use correct words in describe for users_controller_spec.
- Use same syntax for the tests of both controllers for easy understanding.
